### PR TITLE
Three cameras for octomapper

### PR DESCRIPTION
**Pull request by oswinso**
This PR adds the ability for the mapper to map using three cameras.

### CODE REVIEW

**Comment by JasonGibson274** [comment]
You should add a flag to determine what cameras should be used.

**Comment by oswinso** [comment]
Yeah, true. Adding it now.

File: igvc_navigation/src/mapper/mapper.cpp [hunk 1]
```diff
@@ -56,7 +57,12 @@ Mapper::Mapper(ros::NodeHandle& pNh) : ground_plane_{ 0, 0, 1, 0 }
   octomapper_->create_octree(camera_map_pair_);
 
   camera_line_pub_ = pNh.advertise<pcl::PointCloud<pcl::PointXYZ>>("/mapper/camera_lines", 1);
-  camera_projection_pub_ = pNh.advertise<pcl::PointCloud<pcl::PointXYZ>>("/mapper/camera_projection", 1);
+
+  camera_projection_pub_l_ = pNh.advertise<pcl::PointCloud<pcl::PointXYZ>>("/mapper/camera_projection_l", 1);
```
Comment (Alescontrela) [inline igvc_navigation/src/mapper/mapper.cpp]: Expand these topic names out to `camera_projection_left`, etc. for clarity

File: igvc_navigation/src/mapper/mapper.cpp [hunk 2]
```diff
@@ -142,25 +148,43 @@ void Mapper::insertCameraProjection(const pcl::PointCloud<pcl::PointXYZ>::ConstP
...
-    ROS_WARN_STREAM_THROTTLE(1, "camera model not initialized or not using ground filter");
-    return;
+    MapUtils::debugPublishPointCloud(camera_projection_pub_l_, projected_pc, pcl_conversions::toPCL(stamp), "/odom",
+                                     debug_);
+    sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), "mono8", image).toImageMsg();
+    msg->header.stamp = ros::Time::now();
+    msg->header.frame_id = "/left_cam_optical";
+    process_image_pub_.publish(msg);
+  }
+  else if (static_cast<int>(camera) == 1)
```
Comment (Alescontrela) [inline igvc_navigation/src/mapper/mapper.cpp]: why do the center and right cameras not have:
```C++
    sensor_msgs::ImagePtr msg = cv_bridge::CvImage(std_msgs::Header(), "mono8", image).toImageMsg();
    msg->header.stamp = ros::Time::now();
    msg->header.frame_id = "/left_cam_optical";
    process_image_pub_.publish(msg);
```

Comment (oswinso) [reply igvc_navigation/src/mapper/mapper.cpp]: oh rip I forgot to remove this debugging call

File: igvc_navigation/src/mapper/mapper.cpp (hunk 1 above)
Comment (oswinso) [reply igvc_navigation/src/mapper/mapper.cpp]: sure.

File: igvc_navigation/launch/octomap.yaml [hunk 3]
```diff
@@ -15,7 +15,7 @@ probability_model:
         free_space: # Rays that are not detected by the lidar to have occupied points along them
             prob_miss: 0.52
     camera:
-        prob_miss: 0.51
+        prob_miss: 0.5001
```
[the remark this answers is not in the document]
Comment (oswinso) [reply igvc_navigation/launch/octomap.yaml]: I'll make this an issue

File: igvc_navigation/src/mapper/mapper.cpp [hunk 4]
```diff
@@ -176,6 +195,11 @@ void Mapper::processImageFreeSpace(cv::Mat& image) const
   cv::GaussianBlur(image, image, cv::Size(process_image_options_.blur.kernel, process_image_options_.blur.kernel),
                    process_image_options_.blur.sigma, process_image_options_.blur.sigma);
   cv::threshold(image, image, process_image_options_.threshold.threshold, UCHAR_MAX, cv::THRESH_BINARY);
+  cv::Mat element = cv::getStructuringElement(
```
[the remark this answers is not in the document]
Comment (oswinso) [reply igvc_navigation/src/mapper/mapper.cpp]: changed the variable name to be more helpful

**Pull request merged**